src/main/java/tool/designpatterns/verifiers/multiclassverifiers/AdapterVerifier.java

```diff
@@ -0,0 +1,244 @@
+                }
+            } else {
+                for (ClassOrInterfaceType coi : adaptee.getImplementedTypes()) {
+                    if (coit.getNameAsString().equalsIgnoreCase(coi.getNameAsString())) {
+                        return Feedback.getNoChildFeedback(
+                            "The adapter implements the same interface as the adaptee",
+                            new FeedbackTrace(adapter));
+                    }
+                }
+            }
+        }
+
+        return Feedback.getSuccessfulFeedback();
+    }
+
+    /**
+     * A class used to visit nodes in a AST created by JavaParser.
+     */
+
+    private static class MethodDeclarationVisitor
+        extends GenericListVisitorAdapter<Boolean, ClassOrInterfaceDeclaration> {
+
+        private final ClassOrInterfaceDeclaration currentClass;
+
+        public MethodDeclarationVisitor(ClassOrInterfaceDeclaration currentClass) {
+            super();
+            this.currentClass = currentClass;
+        }
+
+        /**
+         * A method for visiting MethodDeclarations in ClassOrInterfaceDeclarations. Visits the
+         * MethodDeclarations and checks if they are overridden. If they are a true feedback is
+         * added to the list, otherwise a false is added.
+         *
+         * @param method  The MethodDeclaration of the method to be checked.
+         * @param adaptee The ClassOrInterfaceDeclaration of the adaptees method to be wrapped.
+         * @return a list of booleans
+         */
+        @Override
+        public List<Boolean> visit(
+            MethodDeclaration method, ClassOrInterfaceDeclaration adaptee) {
+            List<Boolean> resultList = super.visit(method, adaptee);
+            for (AnnotationExpr annotation : method.getAnnotations()) {
+                if (annotation.getNameAsString().equalsIgnoreCase("override")) {
+                    resultList.add(isWrapper(method, currentClass, adaptee));
+                    return resultList;
+                }
+            }
+            resultList.add(Boolean.FALSE);
+            return resultList;
+        }
+
+        /**
+         * A method that checks if a method is called from within another method, i.e. if it is
+         * wrapped.
+         *
+         * @param method  The MethodDeclaration fo the wrapping method.
+         * @param adaptee The ClassOrInterfaceDeclaration to look for the wrapped method in.
+         * @return a boolean, true if it is wrapped, otherwise false.
+         */
+        private Boolean isWrapper(
+            MethodDeclaration method, ClassOrInterfaceDeclaration currentClass,
+            ClassOrInterfaceDeclaration adaptee) {
+            List<Boolean> list = method.accept(new MethodCallVisitor(currentClass), adaptee);
+
+            for (Boolean bool : list) {
+                if (bool) {
+                    return Boolean.TRUE;
+                }
+            }
+            return Boolean.FALSE;
```

Comment:
```suggestion
            return list.stream().anyMatch(e-> e);
```

This code does the same thing and halves the length of the method.